### ISSUE TITLE
ART index creation takes too much memory

### ISSUE THREAD

**Issue by fanvanzh** [duckdb/duckdb#15420]
### What happens?

I used the data of TPCH-100G and add the index to the lineitem table as follows(l_orderkey is bigint, l_linenumber is int)

```
set threads = 4;
set memory_limit = '20GB';
create unique index ii on lineitem(l_orderkey, l_linenumber);
47% ▕████████████████████████████▏    
Out of Memory Error: could not allocate block of size 256.0 KiB (18.6 GiB/18.6 GiB used)
```

The size of the `lineitem` table is only 18GB, but OOM still occurs.
I suspect that the create index loads all data into memory and does not do the necessary buffer culling.

### To Reproduce

INSTALL tpch;
LOAD tpch;
CALL dbgen(sf = 100);
set threads = 4;
set memory_limit = '20GB';
create unique index ii on lineitem(l_orderkey, l_linenumber);

### OS:

linux

### DuckDB Version:

1.1.3

### DuckDB Client:

duckapi-c

### Hardware:

x86-64

### Full Name:

fanvanzh

### Affiliation:

fanvanzh

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a source build

### Did you include all relevant data sets for reproducing the issue?

No - Other reason (please specify in the issue body)

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by mahmoudmostafa0** [comment]
i got the same issue on 1.1.3

**Comment by arjenpdevries** [comment]
Don't you need the 18G for the table itself plus space for the index on the table, hence likely too much for 20G?

**Comment by fanvanzh** [comment]
@taniabogatsch  Could you give me some idea about this question?

**Comment by taniabogatsch** [comment]
Indeed, we also need to materialise some of the table data to compute the index keys.

> I suspect that the create index loads all data into memory and does not do the necessary buffer eviction.

But, to answer your question, indexes are indeed not yet buffer managed. We persist them to disk and later load them lazily, but during index creation, the entire index will be in-memory, including the 'extra' space non-full nodes take/etc. They are also (as a structure) not too memory-friendly.

Here are some PRs to make the indexes buffer-managed, but its WIP without a clear dead/timeline: https://github.com/duckdb/duckdb/pull/18567.

**Comment by tiagozip** [comment]
any update on this?

**Comment by taniabogatsch** [comment]
You can follow the PR "chain" to the latest linked PR - but we haven't converted all index code to use the buffer-managed nodes yet, so it is not yet possible to evict index memory. We'll come back to this issue once indexes are buffer-managed. 👍
